### ISSUE TITLE
ecowitt no loaing, version key error in logs.

### ISSUE THREAD

**Issue by jatgm1** [garbled1/homeassistant_ecowitt#63]
so im not exactly sure whats wrong but ecowitt wont load when i upgrade my home assistant so i kinda stuck on this version. not sure whats wrong. was hoping you could release an update to fix it.

**Comment by flipperkip** [comment]
the same here. updated home assistant core-2021.6.6  version 5.10.17-v8
--
below the error from the home assistant log file:
2021-07-01 22:03:18 ERROR (SyncWorker_3) [homeassistant.loader] The custom integration 'ecowitt' does not have a valid version key (None) in the manifest file and was blocked from loading. See https://developers.home-assistant.io/blog/2021/01/29/custom-integration-changes#versions for more details

Hope this can be fixed. Reinstallation of the integration also didn't work.

Many thanks in advance for any help.

**Comment by jatgm1** [comment]
I'm basically stuck on version 5.5 so idk what to do.

**Comment by jatgm1** [comment]
Gonna have to get another rpi and try to setup another one that's the newest version to see if i can get it to work w/o version key.

**Comment by Joshndroid** [comment]
This is an issue previously mentioned. There is a current pull request awaiting to be merged in https://github.com/garbled1/homeassistant_ecowitt/pull/49

**Comment by GSzabados** [comment]
Temporary fix here:

https://github.com/garbled1/homeassistant_ecowitt/issues/43#issuecomment-853766734

**Comment by flipperkip** [comment]
> so im not exactly sure whats wrong but ecowitt wont load when i upgrade my home assistant so i kinda stuck on this version. not sure whats wrong. was hoping you could release an update to fix it.


I found the solution on the following link:
https://github.com/garbled1/homeassistant_ecowitt/issues/48#issuecomment-856972832

you just have to add the version number to the manifest.json file!
I got it working again by just adding this version number, without reinstalling the whole thing. ;-)
After restarting HA everything is working again.

Good luck

**Comment by jatgm1** [comment]
So where is the json located and where is it added? Like directory wise? And also what do I search for in the file and what do I add?

**Comment by AdeZwart** [comment]
> So where is the json located and where is it added? Like directory wise? And also what do I search for in the file and what do I add?

That would be in `/custom_components/ecowitt/manifest.json`

**Comment by doubleugk** [comment]
tried doing the fix but not working for me :(

HA core-2021.7.2
HACS 1.13.2
ecowitt 0.5
OS 5.11.0-1009-raspi

**Comment by GSzabados** [comment]
> tried doing the fix but not working for me :(

Can you tell with your own words what have you done and what error message do you have in the log?

**Comment by doubleugk** [comment]
I followed the steps, so that I could add ecowitt  in HA integration section, but it gets not listed.

**Comment by GSzabados** [comment]
> I followed the steps, so that I could add ecowitt in HA integration section, but it won't get listed.

Look, I tried to be polite and offer you some help. But if you are not willing to share any details then do not expect  that anyone can guess what has gone wrong.

By the way, have you read the other comments in this issue and have followed the posted links?

**Comment by doubleugk** [comment]
I'm sorry that my answer was not sufficient.

that's what I did, it was mentioned before being helpful

`Restart HomeAssistant
Manually delete all the files and maps in Custom Components / ecowitt
Restart HA
Re-install ecowitt from HACS
Fill in the “version”
Restart HA

And my (working in 2021.6.3) example of the manifest.json file:


{
  "domain": "ecowitt",
  "name": "Ecowitt Weather Station",
  "version": "7.5.1",
  "config_flow": true,
  "documentation": "https://www.home-assistant.io/integrations/ecowitt",
  "requirements": [
    "pyecowitt==0.11"
  ],
  "ssdp": [],
  "zeroconf": [],
  "homekit": {},
  "dependencies": [],
  "codeowners": [
    "***@***.***"
  ]
}
`

**Comment by GSzabados** [comment]
> I'm sorry that my answer was not sufficient.

Do you see any error message in the log regarding Ecowitt? Have you used the integration before, or is it a new installation?

**Comment by doubleugk** [comment]
yes its a complete new installation on a pi4 in a docker container
2021-07-17 16:48:58 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-07-17 16:48:58 WARNING (SyncWorker_1) [homeassistant.loader] We found a custom integration ecowitt which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

**Comment by GSzabados** [comment]
> 2021-07-17 16:48:58 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant

That is standard. It just mean that you have installed a custom component. Anything else? If not, then have you tried to configure it through the Integrations page?

Configuration / Integrations / + Add Integration button and then search for Ecowitt Weather Station 

Select it and configure your settings there, ie. Port and units to use.

**Comment by doubleugk** [comment]
That's the issue - it won't get listed. I thought that when I edit the config file and add the missing version information it will be listed. This has worked for other users.

**Comment by GSzabados** [comment]
> That's the issue - it won't get listed. 

Can you make a screenshot like this?

![Screenshot_20210720-150625_Home Assistant](https://user-images.githubusercontent.com/35445496/126329375-3b1e325f-2f70-4765-8ad3-1d914de46645.jpg)

And try to restart HA and then after it has fully started look into the log again for anything what has ecowitt in it.

**Comment by doubleugk** [comment]
unbelivable - only chatting with u solved the issue ;)
I don't know why, but now the HA integration of ecowitt is possible. 
I've restartet everything so many time with no change.
The only thing what changed today was a long update list from Linux, but I don't know how this could effect HA.
Anyway it's odd but I'm happy :)
Thx very much
